### ISSUE TITLE
AutoScalingGroup CreationPolicy doesnt error

### ISSUE THREAD

**Issue by whoDoneItAgain** [aws-cloudformation/cfn-lint#3598]
### CloudFormation Lint Version

v1.10.1

### What operating system are you using?

windows

### Describe the bug

in the example template, i have a reference to ResourceSignalCount. There isnt a parameter for this but does not throw an error

### Expected behavior

an error is thrown

### Reproduction template

```
AWSTemplateFormatVersion: 2010-09-09

Resources:
  AutoScalingGroup:
    Type: AWS::AutoScaling::AutoScalingGroup
    CreationPolicy:
      ResourceSignal:
        Count: !Ref ResourceSignalCount
        Timeout: PT15M
    Properties:
      MaxSize: 1
      MinSize: 1
```

**Comment by kddejong** [comment]
Some notes for adding support:

* Creation policies are ignored for resources that don't support them (this could be a Warning)

Other fixes:
* One of LaunchTemplate, LaunchConfigurationName, InstanceId or MixedInstancesPolicy is required for `AWS::AutoScaling::AutoScalingGroup`
* At least one Availability Zone or VPC Subnet is required for `AWS::AutoScaling::AutoScalingGroup`
